src/python/pants/backend/python/lint/python_format_target.py

```diff
@@ -29,13 +31,18 @@ class _ConcretePythonFormatTarget:
 @rule
 async def format_python_target(
   wrapped_target: _ConcretePythonFormatTarget, union_membership: UnionMembership
-) -> FmtResults:
-  """This aggregator allows us to have multiple formatters operate over the same Python targets."""
-  results = await MultiGet(
-    Get[FmtResult](PythonFormatTarget, member(wrapped_target.target))
-    for member in union_membership.union_rules[PythonFormatTarget]
-  )
-  return FmtResults(results)
+) -> AggregatedFmtResults:
+  """This aggregator allows us to have multiple formatters safely operate over the same Python
+  targets, even if they modify the same files."""
+  prior_formatter_result_digest: Optional[Digest] = None
+  results: List[FmtResult] = []
+  for member in union_membership.union_rules[PythonFormatTarget]:
+    result = await Get[FmtResult](
+      PythonFormatTarget, member(wrapped_target.target, prior_formatter_result_digest)
```

Comment:
```suggestion
      PythonFormatTarget, member(wrapped_target.target, prior_formatter_result_digest=prior_formatter_result_digest)
```

Being explicit about names of fields when effectively using reflection seems sensible in terms of guaranteeing the API is being used properly.

Reply:
Great idea.